src/widget/translationAnimator.ts

```diff
@@ -107,23 +102,19 @@ export class TranslationAnimator extends Clutter.Actor {
                 }
             });
 
-            for (const actor of initialActors) {
+            for (const actor of this.currentActors) {
                 const p = actor.get_parent();
                 if (p !== null && p !== this.transitionContainer) {
                     p.remove_child(actor);
```

Comment:
This shouldn't happen ideally, though? If an actor is currently being animated, we should never set it to have a parent other than the transition container. Doing that seems like it would lead to terrible behavior?